### PR TITLE
[Artist] Fixed past shows bug & refactored shows lists

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Fix rotation bug in Past Shows
- [x] Make a `ShowsList` that can be used for both large and medium image sizes
- [x] Use new component instead of `LargeList`

Fixes https://github.com/artsy/eigen/issues/1964

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG added.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">The following files do not have tests: <a href="https://github.com/artsy/emission/blob/baa3d9e2646d605528f816336d0184e609717d04/lib/components/artist/shows/index.js">lib/components/artist/shows/index.js</a>, <a href="https://github.com/artsy/emission/blob/baa3d9e2646d605528f816336d0184e609717d04/lib/components/artist/shows/large_list.js">lib/components/artist/shows/large_list.js</a> &amp; <a href="https://github.com/artsy/emission/blob/baa3d9e2646d605528f816336d0184e609717d04/lib/components/artist/shows/medium_list.js">lib/components/artist/shows/medium_list.js</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: lib/components/artist/shows/index.js [hunk 1]
```diff
@@ -47,7 +46,7 @@ class Shows extends React.Component {
       return (
         <View style={{ marginBottom: 20 }}>
           <SerifText style={styles.title}>Current & Upcoming Shows</SerifText>
-          <LargeShowsList shows={shows} />
+          <VariableSizeShowsList showSize={'large'} shows={shows} />
```
Comment (mennenia) [inline lib/components/artist/shows/index.js]: I like this VariableSizeShowsList! ^__^

File: lib/components/artist/shows/variable_size_shows_list.js [hunk 2]
```diff
@@ -3,19 +3,29 @@
...
 
   constructor(props) {
     super(props)
+
+    // The 'showSize' must exist and should be a value this component accommodates
+    if (__DEV__) {
+      const possibleShowSizes = [ 'medium', 'large' ]
+      const providedShowSize = this.props.showSize
+      if (!providedShowSize || possibleShowSizes.indexOf(providedShowSize) < 0) {
+        console.error('[ShowsList] Invalid prop `showSize` of value `' + providedShowSize +
```
Comment (mennenia) [inline lib/components/artist/shows/variable_size_shows_list.js]: ✨

**Comment by mennenia** [comment]
Looks great @sarahscott! Was there a ticket for the bug?

**Comment by sarahscott** [comment]
Yes there is! Just added it to the description

File: lib/components/artist/shows/variable_size_shows_list.js [hunk 3]
```diff
@@ -3,19 +3,29 @@
...
-    height: number
+    height: number,
   }
 
   constructor(props) {
     super(props)
+
+    // The 'showSize' must exist and should be a value this component accommodates
+    if (__DEV__) {
+      const possibleShowSizes = [ 'medium', 'large' ]
```
Comment (alloy) [inline lib/components/artist/shows/variable_size_shows_list.js]: Might want to define an enum by using literal types https://flowtype.org/docs/builtins.html#literal-types (scroll a bit down).

Comment (mennenia) [inline lib/components/artist/shows/variable_size_shows_list.js]: ah yeah, enums would be a good one 👌

Comment (sarahscott) [inline lib/components/artist/shows/variable_size_shows_list.js]: I see! I didn't know we had enums at our disposal. I will give that a shot now.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ### 1.1.1
 
+- Artist VC uses one show component for medium & large lists - sarah
```
Comment (orta) [inline CHANGELOG.md]: this will need to go in a new section, the 1.1.1 release was a week ago

**Comment by alloy** [comment]
Ok, taking this

**Pull request merged**